### PR TITLE
CI: Fix typo setting kubeconfig variable from registry

### PR DESCRIPTION
**Pull request by raybejjani**
This caused some off errors, particularly around how we determine
whether we are in the CI or running with a custom kubeconfig.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9520)

### CODE REVIEW

**Comment by raybejjani** [comment]
test-me-please

**Comment by raybejjani** [comment]
test-missed-k8s

**Comment by raybejjani** [comment]
test-nightly

**Review by ianvernon — APPROVED** [review]

**Comment by raybejjani** [comment]
test-me-please

**Comment by raybejjani** [comment]
test-me-please

**Comment by raybejjani** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

File: test/provision/k8s_install.sh [hunk 1]
```diff
@@ -62,6 +62,7 @@ sudo service serial-getty@ttyS0 start
 if [[ -f  "/etc/provision_finished" ]]; then
     sudo dpkg -l | grep kubelet
     echo "provision is finished, recompiling"
+	$PROVISIONSRC/compile.sh
```
Comment (ianvernon) [inline test/provision/k8s_install.sh]: nit: spacing is off

File: test/provision/k8s_install.sh [hunk 2]
```diff
@@ -307,6 +308,7 @@ if [[ "${HOST}" == "k8s1" ]]; then
     kubectl -n kube-system delete -f ${PROVISIONSRC}/manifest/dns_deployment.yaml || true
     kubectl -n kube-system apply -f ${PROVISIONSRC}/manifest/dns_deployment.yaml
 
+	$PROVISIONSRC/compile.sh
```
Comment (ianvernon) [inline test/provision/k8s_install.sh]: nit: spacing is off

File: test/helpers/kubectl.go [hunk 3]
```diff
@@ -227,6 +233,12 @@ func CreateKubectl(vmName string, log *logrus.Entry) (k *Kubectl) {
 	return k
 }
 
+// EnsureLabels labels nodes in vagrant env. If you are running tests on different cluster you need to label your nodes by yourself
+func (kub *Kubectl) EnsureLabels() {
```
Comment (ianvernon) [inline test/helpers/kubectl.go]: `EnsureLabels` isn't evocative of what this function does. Rename it to something like `LabelNodes`. Is there any documentation that needs to be updated for non-Vagrant environments to label the nodes accordingly given that this step does it automatically for Vagrant environments?

**Comment by raybejjani** [comment]
I can't approve "my own" PR, it seems, but +1 from me.
@ianvernon you seemed interested, as a 3rd party, you would be a great reviewer! :P

File: ginkgo-kubernetes-all.Jenkinsfile [hunk 4]
```diff
@@ -176,7 +176,7 @@ pipeline {
                         KUBECONFIG="${TESTDIR}/vagrant-kubeconfig"
```
Comment (ianvernon) [inline ginkgo-kubernetes-all.Jenkinsfile:176]: Can a description be added to this commit to provide more context? Some of the changes appear unrelated at a first glance; a succinct description would help clean up confusion.

File: test/provision/compile.sh [hunk 5]
```diff
@@ -28,7 +28,7 @@ then
     # Only need to build on one host, since we can pull from the other host.
     if [[ "$(hostname)" == "k8s1" ]]; then
       ./test/provision/container-images.sh cilium_images .
-      if [[ "${CILIUM_IMAGE}" == "" && "${CILIUM_OPERATOR_IMAGE}" == "" ]]; then
+	  if [[ "${CILIUM_IMAGE}" == "" && "${CILIUM_OPERATOR_IMAGE}" == "" && "${CILIUM_REGISTRY}" == "" ]]; then
```
Comment (ianvernon) [inline test/provision/compile.sh:31]: nit: spacing is off.

File: test/helpers/kubectl.go (hunk 3 above)
Comment (raybejjani) [reply test/helpers/kubectl.go]: The how-to-test docs are still a todo for me :/ Once I write them I'll mention this. (but perhaps joestringer already wrote some of this. I don't think it came up for him, though, since it was a single-node test)

**Comment by nebril** [comment]
test-me-please
